gwcs/wcs.py

```diff
@@ -2756,11 +2756,27 @@ def _calc_approx_inv(self, max_inv_pix_error=5, inv_degree=None, npoints=16):
                                 (Shift(crpix[0]) & Shift(crpix[1])))
 
 
-def _poly_fit_lu(xin, yin, xout, yout, degree):
+def _poly_fit_lu(xin, yin, xout, yout, degree, coord_pow=None):
+    # This function fits 2D polynomials to data by writing the normal system
+    # of equations and solving it using LU-decomposition. In theory this
+    # should be less stable than the SVD method used by numpy's lstsq or
+    # astropy's LinearLSQFitter because the condition of the normal matrix
+    # is squared compared to the direct matrix. However, in practice,
+    # in our (Mihai Cara) tests of fitting WCS distortions, solving the
+    # normal system proved to be significantly more accurate, efficient,
+    # and stable than SVD.
+    #
+    # coord_pow - a dictionary used to store powers of coordinate arrays
```

Comment:
doesn't matter. memory caching cannot store/use more than 2x memory used for the pseudo-vandermonde matrix. Just an estimate, for 9th (highest) order SIP it will store 54*npoints**2 values in addition to the same number computed anyway. For 1024 points, it would require about 50-100MB extra depending on longdouble size.

Reply:
Fox an expected typical usage with npoints=256 and longdouble being 128 bits and degree of 5, caching would require 20MB

Reply:
Right. Somehow I was thinking it was for every pixel.